### PR TITLE
Add const qualifier [4397]

### PR DESCRIPTION
**Pull request by julianbermudez**
This pull request adds const qualifier for attributes in `createPublisher` and createSubscriber` functions.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1116/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1086/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1100/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1124/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1094/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1108/)

File: include/fastrtps/Domain.h [hunk 1]
```diff
@@ -67,7 +67,7 @@ class Domain
      * @param listen ParticipantListener Pointer.
      * @return Participant pointer. (nullptr if not created.)
      */
-    RTPS_DllAPI static Participant* createParticipant(ParticipantAttributes &att,
+    RTPS_DllAPI static Participant* createParticipant(ParticipantAttributes& att,
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply include/fastrtps/Domain.h]: Ok, it requires much more modifications. I'm going there...

[the remark this answers is not in the document]
Comment (julianbermudez) [reply include/fastrtps/Domain.h]: Done!

[the remark this answers is not in the document]
Comment (MiguelCompany) [reply include/fastrtps/Domain.h]: Not really, you did it for RTPSDomain, but this one is still pending.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1156/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1126/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1140/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1158/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1128/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1142/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1160/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1129/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1143/)

File: src/cpp/Domain.cpp [hunk 2]
```diff
@@ -204,7 +206,7 @@ Publisher* Domain::createPublisher(Participant *part, const std::string &publish
     return createPublisher(part, publisher_att, listen);
 }
 
-Publisher* Domain::createPublisher(Participant *part, PublisherAttributes &att, PublisherListener *listen)
+Publisher* Domain::createPublisher(Participant *part, const PublisherAttributes &att, PublisherListener *listen)
```
Comment (MiguelCompany) [inline src/cpp/Domain.cpp]: Arguments style.

File: src/cpp/Domain.cpp [hunk 3]
```diff
@@ -252,7 +254,7 @@ Subscriber* Domain::createSubscriber(Participant *part, const std::string &subsc
     return createSubscriber(part, subscriber_att, listen);
 }
 
-Subscriber* Domain::createSubscriber(Participant *part, SubscriberAttributes &att, SubscriberListener *listen)
+Subscriber* Domain::createSubscriber(Participant *part, const SubscriberAttributes &att, SubscriberListener *listen)
```
Comment (MiguelCompany) [inline src/cpp/Domain.cpp]: Arguments style

File: src/cpp/participant/ParticipantImpl.cpp [hunk 4]
```diff
@@ -47,7 +47,7 @@
 using namespace eprosima::fastrtps;
 using namespace eprosima::fastrtps::rtps;
 
-ParticipantImpl::ParticipantImpl(ParticipantAttributes& patt,Participant* pspart,ParticipantListener* listen):
+ParticipantImpl::ParticipantImpl(const ParticipantAttributes& patt, Participant* pspart, ParticipantListener* listen):
```
Comment (MiguelCompany) [inline src/cpp/participant/ParticipantImpl.cpp]: Arguments style

File: src/cpp/participant/ParticipantImpl.cpp [hunk 5]
```diff
@@ -111,7 +111,7 @@ const GUID_t& ParticipantImpl::getGuid() const
     return this->mp_rtpsParticipant->getGuid();
 }
 
-Publisher* ParticipantImpl::createPublisher(PublisherAttributes& att,
+Publisher* ParticipantImpl::createPublisher(const PublisherAttributes& att,
```
Comment (MiguelCompany) [inline src/cpp/participant/ParticipantImpl.cpp]: Arguments style

File: src/cpp/participant/ParticipantImpl.cpp [hunk 6]
```diff
@@ -231,7 +231,7 @@ std::vector<std::string> ParticipantImpl::getParticipantNames() const {
     return mp_rtpsParticipant->getParticipantNames();
 }
 
-Subscriber* ParticipantImpl::createSubscriber(SubscriberAttributes& att,
+Subscriber* ParticipantImpl::createSubscriber(const SubscriberAttributes& att,
```
Comment (MiguelCompany) [inline src/cpp/participant/ParticipantImpl.cpp]: Arguments style

File: src/cpp/participant/ParticipantImpl.h [hunk 7]
```diff
@@ -63,7 +63,7 @@ class ParticipantImpl
     typedef std::vector<t_p_SubscriberPair> t_v_SubscriberPairs;
 
     private:
-    ParticipantImpl(ParticipantAttributes& patt,Participant* pspart,ParticipantListener* listen = nullptr);
+    ParticipantImpl(const ParticipantAttributes& patt,Participant* pspart,ParticipantListener* listen = nullptr);
```
Comment (MiguelCompany) [inline src/cpp/participant/ParticipantImpl.h]: Arguments style

File: src/cpp/participant/ParticipantImpl.h [hunk 8]
```diff
@@ -88,15 +88,15 @@ class ParticipantImpl
      * @param listen Pointer to the listener.
      * @return Pointer to the created Publisher.
      */
-    Publisher* createPublisher(PublisherAttributes& att, PublisherListener* listen=nullptr);
+    Publisher* createPublisher(const PublisherAttributes& att, PublisherListener* listen=nullptr);
```
Comment (MiguelCompany) [inline src/cpp/participant/ParticipantImpl.h]: Arguments style

File: src/cpp/participant/ParticipantImpl.h [hunk 9]
```diff
@@ -88,15 +88,15 @@ class ParticipantImpl
...
 
     /**
      * Create a Subscriber in this Participant.
      * @param att Attributes of the Subscriber
      * @param listen Pointer to the listener.
      * @return Pointer to the created Subscriber.
      */
-    Subscriber* createSubscriber(SubscriberAttributes& att, SubscriberListener* listen=nullptr);
+    Subscriber* createSubscriber(const SubscriberAttributes& att, SubscriberListener* listen=nullptr);
```
Comment (MiguelCompany) [inline src/cpp/participant/ParticipantImpl.h]: Arguments style

File: src/cpp/publisher/PublisherImpl.cpp [hunk 10]
```diff
@@ -35,8 +35,8 @@
 using namespace eprosima::fastrtps;
 using namespace ::rtps;
 
-PublisherImpl::PublisherImpl(ParticipantImpl* p,TopicDataType*pdatatype,
-        PublisherAttributes& att,PublisherListener* listen ):
+PublisherImpl::PublisherImpl(ParticipantImpl* p, TopicDataType*pdatatype,
```
Comment (MiguelCompany) [inline src/cpp/publisher/PublisherImpl.cpp]: Arguments style

File: src/cpp/publisher/PublisherImpl.h [hunk 11]
```diff
@@ -61,7 +61,7 @@ class PublisherImpl
      * Don't use directly, create Publisher using DomainRTPSParticipant static function.
      */
     PublisherImpl(ParticipantImpl* p,TopicDataType* ptype,
-            PublisherAttributes& att,PublisherListener* p_listen = nullptr);
+            const PublisherAttributes& att,PublisherListener* p_listen = nullptr);
```
Comment (MiguelCompany) [inline src/cpp/publisher/PublisherImpl.h]: Arguments style

File: include/fastrtps/Domain.h [hunk 12]
```diff
@@ -67,8 +67,9 @@ class Domain
      * @param listen ParticipantListener Pointer.
      * @return Participant pointer. (nullptr if not created.)
      */
-    RTPS_DllAPI static Participant* createParticipant(ParticipantAttributes &att,
-                                                      ParticipantListener *listen = nullptr);
+    RTPS_DllAPI static Participant* createParticipant(
```
Comment (MiguelCompany) [inline include/fastrtps/Domain.h]: Arguments style.

File: include/fastrtps/Domain.h [hunk 13]
```diff
@@ -79,9 +80,10 @@ class Domain
      * @param listen Pointer to the PublisherListener.
      * @return Pointer to the created Publisher (nullptr if not created).
      */
-    RTPS_DllAPI static Publisher* createPublisher(Participant *part,
-                                                  const std::string &publisher_profile,
-                                                  PublisherListener *listen = nullptr);
+    RTPS_DllAPI static Publisher* createPublisher(
```
Comment (MiguelCompany) [inline include/fastrtps/Domain.h]: Arguments style.

File: include/fastrtps/Domain.h [hunk 14]
```diff
@@ -90,9 +92,10 @@ class Domain
      * @param listen Pointer to the PublisherListener.
      * @return Pointer to the created Publisher (nullptr if not created).
      */
-    RTPS_DllAPI static Publisher* createPublisher(Participant *part,
-                                                  PublisherAttributes &att,
-                                                  PublisherListener *listen = nullptr);
+    RTPS_DllAPI static Publisher* createPublisher(
```
Comment (MiguelCompany) [inline include/fastrtps/Domain.h]: Arguments style.

File: include/fastrtps/Domain.h [hunk 15]
```diff
@@ -103,19 +106,22 @@ class Domain
      * @param listen Pointer to the SubscriberListener.
      * @return Pointer to the created Subscriber (nullptr if not created).
      */
-    RTPS_DllAPI static Subscriber* createSubscriber(Participant *part,
-                                                    const std::string &subscriber_profile,
-                                                    SubscriberListener *listen = nullptr);
+    RTPS_DllAPI static Subscriber* createSubscriber(
```
Comment (MiguelCompany) [inline include/fastrtps/Domain.h]: Arguments style.

File: include/fastrtps/Domain.h [hunk 16]
```diff
@@ -103,19 +106,22 @@ class Domain
...
      * Create a Subscriber in a Participant.
      * @param part Pointer to the participant where you want to create the Publisher.
      * @param att SubscriberAttributes.
      * @param listen Pointer to the SubscriberListener.
      * @return Pointer to the created Subscriber (nullptr if not created).
      */
-    RTPS_DllAPI static Subscriber* createSubscriber(Participant *part,
-                                                    SubscriberAttributes &att,
-                                                    SubscriberListener *listen = nullptr);
+    RTPS_DllAPI static Subscriber* createSubscriber(
```
Comment (MiguelCompany) [inline include/fastrtps/Domain.h]: Arguments style.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1163/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1132/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1146/)

**Pull request merged**